.github/workflows/release-to-pypi.yml

```diff
@@ -0,0 +1,48 @@
+name: Deploy SDK to PyPi
+
+on:
+  pull_request:
+    types: [labeled]
+
+jobs:
+  deploy-to-pypi:
+    runs-on: ubuntu-latest
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'release-sdk-to-pypi') }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - uses: tj-actions/branch-names@v8
+        id: branch-name
+
+      - name: Check branch name is a release branch
+        run: |
+          pattern='(v[0-9]+\.[0-9]+\.[0-9]+|[0-9]+\.[0-9]+\.[0-9]+)'
+
+          branch=${{ steps.branch-name.outputs.current_branch }}
+          echo "current branch is $branch"
+          if [[ ! "$branch" =~ $pattern ]]; then
+            echo "Branch name $branch does not match the pattern. do not use release-sdk-to-pypi on non-release branch"
+            exit 1
+          else
+            echo "branch name $branch is valid, releasing the SDK into pypi"
+          fi
+
+      - name: Python ${{ matrix.python-version }} - Setup Environment
+        uses: ./.github/actions/setup_environment
+        with:
+          python-version: '3.10'
+
+      - name: Poetry build
+        run: |
+          poetry build
+
+#      - name: Publish SDK to PyPi
+#        run: |
+#          poetry publish -u __token__ -p ${PYPI_TOKEN}
```

Comment:
this is on purpose, I will open a PR to get rid of circle-ci and I will un-comment it.